### PR TITLE
fix: settings notices consistency

### LINKED ISSUE #8833: Inconsistent notice style in WooPayments Settings
### Describe the bug
The notice "One or more of your extensions alters checkout fields" should use the correct notice component. The correct notice is shown in the notice just below.

### To Reproduce

1. Go to Settings > WooPayments
2. Scroll to express payments section
3. See bug

### Screenshots
Screenshot 2024-05-14 at 7 27 37 AM

### Expected behavior
The notice should use the `inline-notice` component for a consistent experience in the admin.
The copy can be all on one line and use this simplified text string to clarify the problem.
> Your custom checkout fields may not be compatible with these payment methods. _Learn more_

cc @FangedParakeet @pierorocca

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/8833

#### Changes proposed in this Pull Request

The WooPay compatibility notice is using a different notice component, creating some inconsistency with the notices in the app.

This is the new view:
Screenshot 2024-05-17 at 1 44 31 PM
Screenshot 2024-05-17 at 1 44 36 PM

#### Testing instructions

- I ended up forcing the visibility in these lines: https://github.com/Automattic/woocommerce-payments/blob/693bcb3e6b68a6fe56522d7938c2b832d74a481a/client/settings/express-checkout-settings/woopay-settings.js#L69-L69 https://github.com/Automattic/woocommerce-payments/blob/693bcb3e6b68a6fe56522d7938c2b832d74a481a/client/settings/express-checkout/woopay-item.tsx#L179-L179 https://github.com/Automattic/woocommerce-payments/blob/693bcb3e6b68a6fe56522d7938c2b832d74a481a/client/components/payment-methods-list/payment-method.tsx#L345-L345
- You can also enable the WC Stripe plugin with GooglePay/ApplePay

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8837` or branch name `fix/settings-notices-consistency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/settings-notices-consistency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 96be92d1f75cdc3c693bb934d6a8735f71689ae3
- Build time: 2024-05-23 06:19:41 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/settings-warnings/incompatibility-notice.js [hunk 1]
```diff
@@ -7,48 +7,35 @@ import interpolateComponents from '@automattic/interpolate-components';
...
+import InlineNotice from 'wcpay/components/inline-notice';
 
 const IncompatibilityNotice = ( { message, learnMoreLinkHref } ) => (
-	<Notice
+	<InlineNotice
 		status="warning"
+		icon={ true }
 		isDismissible={ false }
-		className="express-checkout__notice express-checkout__incompatibility-warning"
+		className="express-checkout__notice"
```
Comment (timur27) [inline client/settings/settings-warnings/incompatibility-notice.js]: Just to double-check: we removed the `express-checkout__notice` selector styles and I assume this class name is still needed for the notice to look as expected?

**Review by timur27 — APPROVED** [review]
The incompatibility notice is in line with the rest now! Thanks @frosso 🚀 🚢

File: client/settings/settings-warnings/incompatibility-notice.js (hunk 1 above)
Comment (frosso) [reply client/settings/settings-warnings/incompatibility-notice.js]: Ope, I guess that now that I removed the line in `client/settings/express-checkout-settings/index.scss` this is no longer needed - thank you!
```suggestion
```

**Pull request merged**